### PR TITLE
Adding :Any on **kwargs in function signatures

### PR DESCRIPTION
**Pull request by PaMeirelles**
## PR summary

Marking **kwargs as Any to suppress warnings about "partially unknown" function signatures in pyright
## PR checklist

- [x] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- N/A *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- N/A *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- N/A Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Comment by ksunden** [comment]
Since this changes the generated portions of `pyplot.py`, you will need to rerun `tools/boilerplate.py`.

The failed test is checking whether you get the same output now as is already there, which serves as an indication that that needs to be done.

`pyplot.py` is one of only a small handful of files that we type hint inline, mostly because a large portion of it is generated and having parallel generated code for stubs and implementation was more trouble than it was worth.

File: lib/matplotlib/_docstring.pyi [hunk 1]
```diff
@@ -10,7 +10,7 @@ class Substitution:
     @overload
     def __init__(self, **kwargs: str): ...
     def __call__(self, func: _T) -> _T: ...
-    def update(self, *args, **kwargs): ...  # type: ignore[no-untyped-def]
+    def update(self, *args, **kwargs:Any): ...  # type: ignore[no-untyped-def]
```
Comment (ksunden) [inline lib/matplotlib/_docstring.pyi:13]:
```suggestion
    def update(self, *args, **kwargs:Any): ...
```

I _think_ this type ignore can be removed at this point

**Review by ksunden — CHANGES_REQUESTED** [review]
Style wise, I would prefer `**kwargs: Any` (with the space). We don't actually check style of pyi files in CI (though maybe we should... but that is a task for not this PR) .

pyright warnings wise, I would also suggest doing the same to `*args`, not just `**kwargs`, I guess... (I remain somewhat skeptical that these are actually useful and think that static checkers warning on _calls_ to untyped/partially typed functions is generally iffy at best, especially when the definitions are outside of the module of interest)

But still on balance okay with these being added because it _is_ causing such warnings.

FYI @PaMeirelles, you do not have to close your PR to update it, pushing additional commits/rewriting the history of the PR branch will update it.

File: lib/matplotlib/_docstring.pyi (hunk 1 above)
Comment (PaMeirelles) [reply lib/matplotlib/_docstring.pyi:13]: Thanks for the feedback and the tip about the PR. I will implement your suggestions regarding the style and *args, and also rerun tools/boilerplate.py, and update the Pull Request.

**Comment by PaMeirelles** [comment]
@melissawm  Thanks for the feedback and the tip about the PR. I will implement your suggestions regarding the style and *args, and also rerun tools/boilerplate.py, and update the Pull Request.

**Comment by PaMeirelles** [comment]
I deleted it in an attempt to restart the[ setup process ](https://matplotlib.org/stable/devel/development_setup.html) from the absolute zero, since I was having lots of errors in the installation process. I am also unable to run boilerplate.py, which I think is related to the aforementioned problem. Is this a known issue? Is there any work-around I can employ to circumvent it? Thanks for your patience.

**Comment by tacaswell** [comment]
We currently run boilerplate.py on CI so I do not think we have any known issues with it.

If things are not working for you you should try to sort out how to fix the problems not find work arounds!

**Comment by ksunden** [comment]
@PaMeirelles we recently changed build systems, so the docs you'll want to follow are the developer version: 

https://matplotlib.org/devdocs/devel/development_setup.html

(You linked to stable there. for most developers, the only real change is passing `--no-build-isolation` in to `pip install -e . --no-build-isolation`, noting that you may need to install the dev dependencies first.)

Since the change does not affect the 3.8 branch, it isn't backported there, so the "stable" docs still have the old procedures (rather than being inconsistent on that branch).

**Comment by PaMeirelles** [comment]
@tacaswell, my apologies if my previous message was unclear. I was wondering if there might be a method to execute it without having to configure the editable mode. I don't intend to circumvent the proper process; I simply seek an alternative approach. I should also mention that I've invested a substantial amount of time attempting to resolve the issue independently, yet regrettably, I haven't had any success. Nevertheless, I remain committed to contributing and that's why I inquired about a potential alternative solution.

@ksunden The documentation you linked solved my previous error, but left me stuck with another one: 
`Downloading freetype-2.6.1 source from https://download.savannah.gnu.org/releases/freetype/freetype-old/freetype-2.6.1.tar.gz
  <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1002)>`

I attempted the [listed procedures ](https://matplotlib.org/devdocs/devel/dependencies.html#dependencies) regarding manual installation of FreeType but they also haven't worked. I will keep trying to solve it. Thank you once again for your patience and guidance.

**Comment by ksunden** [comment]
That looks like either a network failure or something of that sort...

I'll note we used to use a different download link (pre-meson) by default (and actually allow fallback to that link). I'll also note that it redirects for me to a mirror site, but does download for me (but makes debugging rather hard, as it appears to be locale-dependent)

Hopefully this is only temporary and whichever certificate needs updating is updated swiftly (like I said, the one I got redirected to worked and had a valid certificate)

**Comment by ksunden** [comment]
Here is the pre-meson version of the download URLS (LOCAL_FREETYPE_VERSION="2.6.1" by default)

https://github.com/matplotlib/matplotlib/blob/32402754d9b2c5b997a197f12fec80befb813879/setupext.py#L608C9-L620C10

Here is the new version of that being specified:

https://github.com/matplotlib/matplotlib/blob/35db7dd8296f1897d06206a495629b5f5f7e8d4a/subprojects/freetype-2.6.1.wrap#L2

Note that there is only one link and it is actually the last of the links provided by the old system....

You may be able to workaround it by changing the link for the `sourceforge` version or something (I wouldn't commit that change, but it might be a way to get it building)

**Comment by ksunden** [comment]
Going to https://download.savannah.gnu.org/mirror-check/ may also be elucidating, as it will show you which mirror you are likely to get (and then you can perhaps look into the certificate, but if just changing that link works, it is probably easiest.

**Comment by melissawm** [comment]
@PaMeirelles - if you need help, feel free to reach out on [our chat platform (gitter/Matrix)](https://app.gitter.im/#/room/#matplotlib_matplotlib:gitter.im), I'd be happy to try and debug these problems!

**Comment by PaMeirelles** [comment]
@ksunden your solution of changing the link worked! Thank you so much! I will try to finish the setup process and then make the changes.

@melissawm  Thank you so much for offering your help! I really appreciate it. I think I've managed to solve it for now, thanks to the fix above, but thanks again for being willing to assist.

**Pull request closed without merging**